src/org/labkey/test/util/AuditLogHelper.java

```diff
@@ -69,6 +77,112 @@ public DataRegionTable goToAuditEventView(String eventType)
         return new DataRegionTable("query", _wrapper);
     }
 
+    /**
+     * Get the audit logs from the LabKey server filtered to the given project.
+     *
+     * @param containerPath Path of the LK container to use for the select command.
+     * @param auditEventName Name of the audit event to filter on. Example 'SamplesWorkflowAuditEvent'.
+     * @param columnNames The name of the columns to return.
+     * @param filters The filters to be applied
+     * @return A rowResponse with the query logs.
+     * @throws IOException Can be thrown by the SelectRowsCommand.
+     * @throws CommandException Can be thrown by the SelectRowsCommand.
+     */
+    public SelectRowsResponse getAuditLogsFromLKS(String containerPath, String auditEventName, List<String> columnNames,
+                                                         List<Filter> filters, @Nullable Integer maxRows) throws IOException, CommandException
+    {
+        SelectRowsCommand cmd = new SelectRowsCommand("auditLog", auditEventName);
+        cmd.setColumns(columnNames);
+        cmd.addFilter("ProjectId/Name", _wrapper.getCurrentProject(), Filter.Operator.EQUAL);
+        filters.forEach(cmd::addFilter);
+        if (maxRows != null)
+            cmd.setMaxRows(maxRows);
+        return cmd.execute(_connectionSupplier.get(), containerPath);
+    }
+
+    /**
+     * Check the number of diffs in the audit event. This is a helper function to check the number of diffs in the
+     * newRecordMap for an audit entry. If a transactionId is provided, it will check all rows for that
+     * transactionId. If no transactionId is provided, it will check just the latest row.
+     */
+    public void checkTimelineAuditEventDiffCount(String containerPath, List<Integer> expectedDiffCounts) throws IOException, CommandException
+    {
+        checkTimelineAuditEventDiffCount(containerPath, getAuditEventNameFromURL(), expectedDiffCounts);
+    }
+    public void checkTimelineAuditEventDiffCount(String containerPath, String auditEventName, List<Integer> expectedDiffCounts) throws IOException, CommandException
+    {
+        Integer maxRows = expectedDiffCounts.size();
+        List<Map<String, Object>> events = getAuditLogsFromLKS(containerPath, auditEventName, List.of("NewRecordMap"), Collections.emptyList(), maxRows).getRows();
```

Comment:
Should you assert the the events count matches the expectedDiffsCounts?

Reply:
yup, added.